### PR TITLE
Couple of perf tweaks for Zig and C++

### PR DESCRIPTION
**Pull request by surma**
I tried to make this easier to review commit-by-commit.

Couldn’t squeeze a lot of perf out of C++ version in the time limit I set myself. I suspect the JSON parser might not be very good, tbh.

But here are the perf results.

### Before
Screenshot 2023-03-21 at 13 31 27
### After
Screenshot 2023-03-21 at 13 50 12

### CODE REVIEW

File: app/extensions/order-discount-zig/src/main.zig [hunk 1]
```diff
@@ -76,38 +52,24 @@ pub fn main() u8 {
...
-        };
-        if (bytesRead == 0) {
-            break; // No more data
-        }
-        inputStr.appendSlice(buffer[0..bytesRead]) catch |err| {
-            std.debug.print("Error reading function input: {!}",.{err});
-            return 1;
-        };
-    }
+    var buffer: [3 * 1024]u8 = undefined;
```
Comment (surma) [inline app/extensions/order-discount-zig/src/main.zig:55]: Little trick as our input is limited to max 3K afaik

Comment (nickwesselman) [reply app/extensions/order-discount-zig/src/main.zig:55]: Input actually isn't limited today, but 3K is probably still a better default.

Comment (nickwesselman) [reply app/extensions/order-discount-zig/src/main.zig:55]: An actually, I think this is causing zig to fail on the large cart, thus the abnormally slow instruction count there.

**Comment by nickwesselman** [comment]
Huge improvements, made some additional changes to Zig and merged all this in #11 ... Thank you!!

**Pull request merged**